params/json.libevm_test.go

```diff
@@ -144,3 +145,60 @@ func TestChainConfigJSONRoundTrip(t *testing.T) {
 		})
 	}
 }
+
+func Test_UnmarshalChainConfig(t *testing.T) {
```

Comment:
I suspect that the existing test cases will be sufficient if you refactor as I recommended above. That will also give us a greater degree of confidence between the situations where extras are explicitly registered vs merely passed to the function.

Reply:
They are for the happy path for both the marshal and unmarshal functions. However, I prefer to have unit tests for each marshal and umarshal functions, on top of the existing test, to test unhappy paths/corner cases, especially given the `any`/generic thingies which are prone to error.